### PR TITLE
Change Plot recommendations

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Plots.jl is in the top 20 packages [according to the Julia Pulse (ranking via stars)](http://pkg.julialang.org/pulse.html). Because of its user-friendliness and high amount of development activity (with help always available in the Gitter channel), I believe it should be right here, front and center, on the Julia webiste. On the otherhand, most people will not be plotting directly into Compose.jl, and so I believe showing that as a "plotting package" confuses most newcommers (as evidenced by the questions we get in the Gitter/IRC about it. Plus the link was broken).

So I added a part for Plots.jl and removed the Part about Compose.jl. I think Plots.jl should come by default on JuliaBox to match this as well.

### CODE REVIEW

**Comment by ViralBShah** [comment]
I think the order should be `Plots.jl` with its own description, and then other packages afterwards. Should we also include Vega.jl and GLVisualize.jl? Compose.jl should perhaps completely go from here.

Cc @shashi

File: downloads/index.md [hunk 1]
```diff
@@ -13,10 +13,7 @@ We provide several ways for you to run Julia:
 
 Plotting capabilities are provided by external packages such as
 [PyPlot.jl](https://github.com/stevengj/PyPlot.jl) and
-[Gadfly.jl](http://gadflyjl.org). You can also do much more with
-[Compose.jl](http://composejl.org), a vector graphics library for
-Julia. Look at the [plotting instructions](plotting.html) to install a
```
Comment (Evizero) [inline downloads/index.md]: did your remove the link to plotting.html on purpose?

Comment (ChrisRackauckas) [inline downloads/index.md]: No. Fixed that. Also fixed the grammar error on the next line: "all OF these ..."

**Comment by ChrisRackauckas** [comment]
plotting.html should be updated to match the recommended packages as well.

**Comment by SimonDanisch** [comment]
I think GLVisualize shouldn't be on there. Maybe GLPlot, or GLVisualize in form of a Plots backend. In the end, GLVisualize should be more like Compose ;)
I'm a bit sorry, my release got a bit out of control. I decided to include a refactoring into my release to solve one of the last standing issues in a very elegant way. This means that the almost finished release of the all new GLPlot and Plots backend moved a bit back.
On the bright side, this brings a WebGL backend, great IJulia/Atom support and higher image quality way closer :)
By the way, I discussed with a Berlin startup to use Vega as a frontend language for e.g. GLVisualize.
They might be interested to help out with that. It'll be a nice way to define interactive graphics in a portable way. You liked Vega quite a bit too, right @shashi ? I must admit that I haven't understood it until recently, but it seems to solve a few things quite elegantly and seems to be well adopted.

**Comment by SimonDanisch** [comment]
Actually, let me try to backport the features needed for GLPlot and the Plots-Backend release, before I get deeper into the refactor wormhole. I need to learn this, and that way people will have more time to test things out!

File: downloads/index.md [hunk 2]
```diff
@@ -13,11 +13,8 @@ We provide several ways for you to run Julia:
...
-plotting package. If you are using JuliaBox, all these plotting
-packages are pre-installed.
+[Gadfly.jl](http://gadflyjl.org). A package which integrates most of Julia's plotting backends into one convenient and well-documented API is [Plots.jl](https://github.com/tbreloff/Plots.jl). Look at the [plotting instructions](plotting.html) to install a
```
Comment (tkelman) [inline downloads/index.md]: wrap long lines

**Comment by ViralBShah** [comment]
I am also seeing GR.jl becoming popular.

**Comment by ChrisRackauckas** [comment]
I think one of the big reasons why it's becoming popular is that it's a good backend for Plots. It has a lot of the features of PyPlot, without the cruft (and installation problems) of PyPlot. It still seems to be on its way up.

Vega may be another good candidate here. 

How do we start choosing?

**Comment by Evizero** [comment]
GR.jl certainly deserves recognition beyond "just" being a Plots.jl backend

**Comment by ChrisRackauckas** [comment]
It certainly does. I didn't intend for that to read as though it's "just" a Plots backend, but I think it's a big component of it. Does it make the "the cut" for being featured on the website? Should all of this be put to a longer plotting.html page?

**Comment by ViralBShah** [comment]
I generally feel like this is a place to list all easy to install and currently maintained graphics packages - with some guidance on who might find which ones easy to use and a couple of important features that are defining and make the packages different from the others.

Sorry if this started as a simple exercise and is getting complicated. I am happy to merge something simple and edit the rest later.

**Comment by ViralBShah** [comment]
In the ideal world, this page would also have screenshots of each - perhaps in a future PR.

**Comment by ViralBShah** [comment]
Cc @tbreloff @jheinen @spencerlyon2

**Comment by spencerlyon2** [comment]
Thanks for bringing me into the conversation. I'd be happy to write a blurb and post screenshots of PlotlyJS.jl if it helps. @ViralBShah please let me know if you need me to help here. 

I also think that it should have its own section (outside of a Plots.jl backend) because as far I know PlotlyJS.jl offers some features you can't get in any of the other packages listed here (e.g. maps).

**Comment by ChrisRackauckas** [comment]
We might as well use this PR to really clarify the current state of Julia plotting. People come to the Gitter/IRC confused all the time (a common question is whether there will be a plotting package in base). I think we should really work out what's going on, and write a good explanation in plotting.html (with a quick two sentence summary on the downloads page).

[This right now is the most comprehensive summary of plotting in Julia](https://juliaplots.github.io/backends/). But it's still not complete because it leaves out things like Vega. Should we just kind of list everything? If we include many plotting packages, how to we order them?

**Comment by Evizero** [comment]
As the author of UnicodePlots.jl I want to point out that I am perfectly fine with leaving out UnicodePlots from the page. It is not that much of a typical plotting use-case I would say. And also maybe too much information can also be overwhelming to new users, no?

> I generally feel like this is a place to list all easy to install and currently maintained graphics packages 

I agree with that. It seems most logical that the official page should point to the packages that are most actively maintained while also having the most to offer. Plots and GR most certainly fall into that category. I know too little about the others to give an opinion on those

**Comment by ChrisRackauckas** [comment]
How much is too much? Easy to install and currently maintained would be:

- Plots
- PyPlot
- GR
- Plotly
- PlotlyJS
- Vega
- PGFPlots

I pretty consistently use all but Vega in that list (since it's the one that's not part of Plots.jl...) and they all seem to pass the "easy to install and currently maintained" test (except maybe PGFPlots. I had some install troubles. But that could just be me.) The exclusion list is currently:

- GLVisualize
- Unicode Plots
- Compose

We have to now address the elephant in the room: Gadfly? It's pretty much front and center everywhere, and has the most stars of any repo, but its development activity has basically stalled and its tests haven't passed for months. Does it make the list? Is it even v0.5 compatible, and is any one testing / updating it? If not, we should really reconsider having Gadfly all over the website with v0.5 set to become the new version.

**Comment by Evizero** [comment]
Gadfly still seems to be the Escher plotting choice.

**Comment by ChrisRackauckas** [comment]
But that doesn't mean that it matches the "easy to install and currently maintained" criteria. Here's Gadfly in a new v0.5-rc build:

```julia
julia> using Gadfly
INFO: Precompiling module Gadfly...

WARNING: deprecated syntax "Base.(:*)".
Use "Base.:*" instead.

WARNING: deprecated syntax "Base.(:*)".
Use "Base.:*" instead.
WARNING: symbol is deprecated, use Symbol instead.
 in depwarn(::String, ::Symbol) at .\deprecated.jl:64
 in symbol(::String, ::Vararg{String,N}) at .\deprecated.jl:30
 in include_from_node1(::String) at .\loading.jl:426 (repeats 2 times)
 in macro expansion; at .\none:2 [inlined]
 in anonymous at .\<missing>:?
 in eval(::Module, ::Any) at .\boot.jl:234
 in process_options(::Base.JLOptions) at .\client.jl:239
 in _start() at .\client.jl:318
while loading C:\Users\Chris\.julia\v0.5\Compose\src\form.jl, in expression starting on line 6
ERROR: LoadError: LoadError: Not an `in` expression
 in include_from_node1(::String) at .\loading.jl:426 (repeats 2 times)
 in macro expansion; at .\none:2 [inlined]
 in anonymous at .\<missing>:?
 in eval(::Module, ::Any) at .\boot.jl:234
 in process_options(::Base.JLOptions) at .\client.jl:239
 in _start() at .\client.jl:318
while loading C:\Users\Chris\.julia\v0.5\Compose\src\form.jl, in expression starting on line 188
while loading C:\Users\Chris\.julia\v0.5\Compose\src\Compose.jl, in expression starting on line 80
ERROR: LoadError: Failed to precompile Compose to C:\Users\Chris\.julia\lib\v0.5\Compose.ji
 in compilecache(::String) at .\loading.jl:505
 in require(::Symbol) at .\loading.jl:337
 in include_from_node1(::String) at .\loading.jl:426
 in macro expansion; at .\none:2 [inlined]
 in anonymous at .\<missing>:?
 in eval(::Module, ::Any) at .\boot.jl:234
 in process_options(::Base.JLOptions) at .\client.jl:239
 in _start() at .\client.jl:318
while loading C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl, in expression starting on line 9
ERROR: Failed to precompile Gadfly to C:\Users\Chris\.julia\lib\v0.5\Gadfly.ji
 in compilecache(::String) at .\loading.jl:505
 in require(::Symbol) at .\loading.jl:364
```

If I checkout master on Gadfly and Compose, I still get this:

```julia
julia> using Gadfly

julia> plot(rand(4,4))
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
in open_file at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
in open_file at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
in open_file at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
in open_file at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl
WARNING: Base.OS_NAME is deprecated.
  likely near no file:0
in open_file at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl
WARNING: symbol is deprecated, use Symbol instead.
 in depwarn(::String, ::Symbol) at .\deprecated.jl:64
 in symbol(::String, ::Vararg{String,N}) at .\deprecated.jl:30
 in apply_scale(::Gadfly.Scale.DiscreteScale, ::Array{Gadfly.Aesthetics,1}, ::Gadfly.Data, ::Vararg{Gadfly.Data,N}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\scale.jl:421
 in apply_scales(::Iterators.Distinct{Base.ValueIterator{Dict{Symbol,Gadfly.ScaleElement}}}, ::Array{Gadfly.Aesthetics,1}, ::Gadfly.Data, ::Vararg{Gadfly.Data,N}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\scale.jl:39
 in apply_scales(::Iterators.Distinct{Base.ValueIterator{Dict{Symbol,Gadfly.ScaleElement}}}, ::Gadfly.Data) at C:\Users\Chris\.julia\v0.5\Gadfly\src\scale.jl:59 in render_prepare(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:721
 in render(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:799 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/html")}, ::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1061 in macro expansion at .\multimedia.jl:143 [inlined]
 in display(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1013
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at .\REPL.jl:149
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at .\REPL.jl:136
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at .\REPL.jl:646
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at .\LineEdit.jl:1570
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at .\REPL.jl:897
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##808#809) at .\REPL.jl:182
 in _start() at .\client.jl:360
while loading no file, in expression starting on line 0
WARNING: symbol is deprecated, use Symbol instead.
 in depwarn(::String, ::Symbol) at .\deprecated.jl:64
 in symbol(::String, ::Vararg{String,N}) at .\deprecated.jl:30
 in apply_statistic(::Gadfly.Stat.TickStatistic, ::Dict{Symbol,Gadfly.ScaleElement}, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics) at C:\Users\Chris\.julia\v0.5\Gadfly\src\statistics.jl:764
 in apply_statistics(::Array{Gadfly.StatisticElement,1}, ::Dict{Symbol,Gadfly.ScaleElement}, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics) at C:\Users\Chris\.julia\v0.5\Gadfly\src\statistics.jl:38
 in render_prepare(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:746
 in render(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:799 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/html")}, ::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1061 in macro expansion at .\multimedia.jl:143 [inlined]
 in display(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1013
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at .\REPL.jl:149
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at .\REPL.jl:136
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at .\REPL.jl:646
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at .\LineEdit.jl:1570
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at .\REPL.jl:897
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##808#809) at .\REPL.jl:182
 in _start() at .\client.jl:360
while loading no file, in expression starting on line 0
WARNING: symbol is deprecated, use Symbol instead.
 in depwarn(::String, ::Symbol) at .\deprecated.jl:64
 in symbol(::String, ::Vararg{String,N}) at .\deprecated.jl:30
 in apply_statistic(::Gadfly.Stat.TickStatistic, ::Dict{Symbol,Gadfly.ScaleElement}, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics) at C:\Users\Chris\.julia\v0.5\Gadfly\src\statistics.jl:768
 in apply_statistics(::Array{Gadfly.StatisticElement,1}, ::Dict{Symbol,Gadfly.ScaleElement}, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics) at C:\Users\Chris\.julia\v0.5\Gadfly\src\statistics.jl:38
 in render_prepare(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:746
 in render(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:799 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/html")}, ::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1061 in macro expansion at .\multimedia.jl:143 [inlined]
 in display(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1013
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at .\REPL.jl:149
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at .\REPL.jl:136
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at .\REPL.jl:646
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at .\LineEdit.jl:1570
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at .\REPL.jl:897
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##808#809) at .\REPL.jl:182
 in _start() at .\client.jl:360
while loading no file, in expression starting on line 0
WARNING: Base.ASCIIString is deprecated, use String instead.
  likely near no file:0
in render at C:\Users\Chris\.julia\v0.5\Gadfly\src\geom/point.jl
Error showing value of type Gadfly.Plot:
ERROR: The following aesthetics are required by Geom.point but are not defined: y, x

 in assert_aesthetics_defined(::String, ::Gadfly.Aesthetics, ::Symbol, ::Vararg{Symbol,N}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\aesthetics.jl:160
 in render(::Gadfly.Geom.PointGeometry, ::Gadfly.Theme, ::Gadfly.Aesthetics) at C:\Users\Chris\.julia\v0.5\Gadfly\src\geom\point.jl:31
 in render(::Gadfly.Geom.PointGeometry, ::Gadfly.Theme, ::Gadfly.Aesthetics, ::Array{Gadfly.Aesthetics,1}, ::Array{Gadfly.Data,1}, ::Dict{Symbol,Gadfly.ScaleElement}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\geometry.jl:48
 in (::Gadfly.##86#88{Dict{Symbol,Gadfly.ScaleElement}})(::Tuple{Gadfly.Layer,Gadfly.Aesthetics,Array{Gadfly.Aesthetics,1},Array{Gadfly.Data,1},Gadfly.Theme}) at .\<missing>:0
 in collect(::Base.Generator{Zip{Array{Gadfly.Layer,1},Zip{Array{Gadfly.Aesthetics,1},Zip{Array{Array{Gadfly.Aesthetics,1},1},Base.Zip2{Array{Array{Gadfly.Data,1},1},Array{Gadfly.Theme,1}}}}},Gadfly.##86#88{Dict{Symbol,Gadfly.ScaleElement}}}) at .\array.jl:271
 in #render_prepared#84(::Bool, ::Bool, ::Function, ::Gadfly.Plot, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics, ::Array{Gadfly.Aesthetics,1}, ::Array{Array{Gadfly.StatisticElement,1},1}, ::Array{Array{Gadfly.Aesthetics,1},1}, ::Array{Array{Gadfly.Data,1},1}, ::Dict{Symbol,Gadfly.ScaleElement}, ::Array{Gadfly.GuideElement,1}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:864
 in render_prepared(::Gadfly.Plot, ::Gadfly.Coord.Cartesian, ::Gadfly.Aesthetics, ::Array{Gadfly.Aesthetics,1}, ::Array{Array{Gadfly.StatisticElement,1},1}, ::Array{Array{Gadfly.Aesthetics,1},1}, ::Array{Array{Gadfly.Data,1},1}, ::Dict{Symbol,Gadfly.ScaleElement}, ::Array{Gadfly.GuideElement,1}) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:857
 in render(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:803 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/html")}, ::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1061 in macro expansion at .\multimedia.jl:143 [inlined]
 in display(::Gadfly.Plot) at C:\Users\Chris\.julia\v0.5\Gadfly\src\Gadfly.jl:1013
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at .\REPL.jl:149
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at .\REPL.jl:136
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at .\REPL.jl:646
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at .\LineEdit.jl:1570
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at .\REPL.jl:897
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##808#809) at .\REPL.jl:182
 in _start() at .\client.jl:360
```

Even the most simple commands don't work. Unless someone is willing to take it up by themselves to fix it by the v0.5 release, I don't think it should be featured anymore.

**Comment by oxinabox** [comment]
while @Evizero says:
> As the author of UnicodePlots.jl I want to point out that I am perfectly fine with leaving out UnicodePlots from the page. It is not that much of a typical plotting use-case I would say

I kinda disagree (but not too strongly). 
Since Gadfly stopped working with 0.5, I've been using UnicodePlots (Its great).
It is the only plotting option that works out of the box, in the REPL, over SSH.

And doing things in the REPL over SSH is a pretty typical early game use case, no?

**Comment by asbisen** [comment]
My 2cents. 

I use `PlotlyJS` on a daily basis and love it. `PlotlyJS` coupled with [plotly.js documentation](https://plot.ly/javascript/reference/) make it very easy to visualize most of the data I deal with. 

I have tried `Plots.jl` and it looks very promising but I would suggest keeping PlotlyJS in the list as there is a big community of users (coming from Python, R, Matlab) already familiar with Plotly's API and for them Plots.jl would be a new interface to learn.

**Comment by rsrock** [comment]
FWIW, Gadfly works for me, if you call it like this:

```
plot(x=rand(4), y=rand(4))
```

I do get an impressive wall of deprecation warnings, of course.  Also, `TerminalExtensions.jl` got angry, so no plot in my iTerm session, but I did get the plot in the browser window automatically.

**Comment by ChrisRackauckas** [comment]
That worked (with wall of warnings). So it's not as bad as I thought. But still, is there anyone stepping up to maintain it? Compose is still getting work (which is likely why Gadfly somewhat works on v0.5), but there's really nothing going on in Gadfly. For almost half a year now you can go through issues on Plots.jl, Compose, etc. and you can see all of the issues on Gadfly asking if anyone's going to work on it or fork it. 

If it's actually going to be forked, someone should really do that now, fix the dep warnings, and get something good for v0.5-rc. If no one is willing to do this, I really think the response should not be "wait for someone to fork it, then it'll be back in business" and instead be "no one is working on it. It's pretty much deprecated unless you want to work on it". For a package that's recommended on the Julia website, that's been the response for too long for a language that's changing too fast.

**Comment by tbreloff** [comment]
Thanks for starting the conversation. I think we should consider avoiding
all mention of individual packages on the main page, fixing up the plotting
page with honest and complete information, and adding clear links to that
page.

I think that potential new users which see a list of 10 packages will be
overwhelmed and assume that plotting just "isn't mature yet" and it will
turn them away.  We want to provide simplicity and suggestion. If we decide
to list packages, it should be in the form of this:
juliaplots.github.io/backends. Summary, comparison, and suggestion, not
just listing.

Tl;dr don't overwhelm a new user. Guide them, and champion the strengths of
the current plotting ecosystem.

**Comment by ChrisRackauckas** [comment]
I totally agree, which is why I put the list together based off of what has been mentioned. It's a long list, and I think it's overwhelming (and it frightens users in the Gitter all the time). But for any way we choose only some packages, we better have a clear reason why. 

Not featuring any plotting package on the main downloads page and having them all in plotting.html is a good idea. However, how do we structure that? I agree that summaries and comparisons are the way to go, but is that backend list not already overwhelming to new users (and not even complete)?

**Comment by lobingera** [comment]
One note on Gadfly:
I might remember this wrong, but wasn't the idea of https://github.com/JuliaPlot the collaboration on plotting tools? Gadfly could and should be (after getting an OK from the original author) forked there and just updated to a recent language level.

**Comment by ChrisRackauckas** [comment]
That was an idea about a year ago, and that org hasn't been touched since. But people have still been throwing around the idea that "someone" is going to fork Gadfly and start working on it there, but it's been quite a long time and no one has.

**Comment by rsrock** [comment]
I'm not sure it's necessary to fork Gadfly, there's a whole bunch of core Julialang contributors that have write access to that repo.  I think that the wall of deprecations will disappear over time.  This is pretty much how it worked out after the switch from 0.3 to 0.4.  Remember, package authors have been asked to check out 0.5 only about a week ago, and many of the folks with write access to Gadfly have bigger fish to fry at the moment.  I'd recommend patience.

**Comment by lobingera** [comment]
@rsrock 
> there's a whole bunch of core Julialang contributors that have write access to that repo.

Yourself?
While it might look as the pragmatic solution because some contributors show up in various places to bring julia forward anyway, putting more work on their shoulders might not give us long term gain.

**Comment by ViralBShah** [comment]
@dcjones keeps adding whoever needs commit access.

**Comment by ViralBShah** [comment]
For now, I am merging this PR, but the discussion is by no means complete. I will myself rework this page once we release 0.5 and all the packages get up.

**Comment by tbreloff** [comment]
> I might remember this wrong, but wasn't the idea of https://github.com/JuliaPlot the collaboration on plotting tools? 

The discussions in the JuliaPlot org are what prompted me to start Plots.  I built a "we can do this better" prototype and it grew from there.  I don't think anyone really has interest in continuing that organization, and I feel like the org should be removed.

> Gadfly could and should be (after getting an OK from the original author) forked there and just updated to a recent language level.

Sure, but I'm not sure who really wants to work on it.  It's not just a problem of people being busy... the problem is that Gadfly is usually not the best tool for the job.  It's not the fastest, or the easiest, or the most interactive.  It's heavyweight.  It doesn't support 3D. Subplot support is limited to hstack/vstack and facets.  It's difficult to extend/improve.  (etc)

I think there are pieces of Gadfly that are excellent, but as a complete package it is far behind the other great packages that have already been listed.  Unless someone steps up and says "I really care about this package and I'm willing to spend the time to improve/maintain it" we should consider Gadfly as "unmaintained" and tell new users about this status.

**Comment by rsrock** [comment]
> @rsrock
> 
> there's a whole bunch of core Julialang contributors that have write access to that repo.
> Yourself?

Nope.

> While it might look as the pragmatic solution because some contributors show up in various places to bring julia forward anyway, putting more work on their shoulders might not give us long term gain.

I don't think I was calling for more work on anyone's shoulders when I asked for "patience".  Kind of the opposite, in fact.  I was pointing out that a fork is not needed, because Gadfly is by no means abandoned.  If activity is low, that's because it works pretty well on the Julia 0.4 series.  I use it all the time there.

I took a look around to see if I can contribute any PRs on this, and it turns out that many of them have already been filed.  Most of the issues are in Gadfly's dependencies.  The big one, as far as I can tell, is in Compose:  dcjones/Compose.jl#211.  That PR is four days old.  Keno has a branch of Gadfly that cleans up many of the problems, and I've picked it up and cleaned up a few more things.  But that work will be much easier when Compose et al. are all fixed.

**Comment by rsrock** [comment]
@tbreloff, I like the work you've done on Plots.jl, it's a heroic effort.  However, I don't think we should be in a hurry to shut down one of the most popular Julia packages.  Much of your criticism of Gadfly would also apply to ggplot2, but ggplot2 is great stuff.  I use both Gadfly and ggplot2 all the time.

**Comment by jheinen** [comment]
@tbreloff : I fully agree! Maintenance is really important, in particular in an environment like Julia, which is getting better every day. Actually an easy installation is one of the most important features (many) users expect. That's why we will continue to work on improving the GR installation process. It seems that we have already managed this for macOS X and Windows - but the Linux packages are still WIP. I still expect the installation to be as easy and fast as for the Julia environment itself. Once we have solved these problems, we would be pleased to further improve GR as one of the favourite Plots backend and - probably - as an alternative for former MATLAB users. In the end, an excellent languages such as Julia should not require or depend on a graphics library of another REPL language, although Matplotlib is a great (and still the leading) package - that's out of question.

**Comment by tbreloff** [comment]
>  I don't think we should be in a hurry to shut down one of the most popular Julia packages

To be clear, I'm not saying "shut this down, no one should use this!".   I want new users to be well informed that, despite its apparent popularity, it is not the package-of-choice for many experienced users, and that it is not officially supported by the core Julia team.  The language on the website implies that Gadfly is the "standard library plotting package", and I think many get a sour impression of Julia because of it.

**Comment by tlnagy** [comment]
> I don't think we should be in a hurry to shut down one of the most popular Julia packages. Much of your criticism of Gadfly would also apply to ggplot2, but ggplot2 is great stuff. I use both Gadfly and ggplot2 all the time.

@rsrock I too, really like Gadfly and I'm using it in a publication that I'm working on atm. I'm interested in keeping it afloat and in good working condition.

**Comment by JeffreySarnoff** [comment]
a tag indexable `dictionary` of visual exemplars and their pkg reqs would make choosing easier to picture for newcomers
